File: 706. Design HashMap.py
##ss
class MyHashMap:

    def __init__(self):
        self.dicts = {}
        

    def put(self, key: int, value: int) -> None:
        
        self.dicts[key] = value
        

    def get(self, key: int) -> int:
        if key in self.dicts.keys():
            return self.dicts[key]
        
        else:
            return -1
        

    def remove(self, key: int) -> None:
        self.dicts.pop(key,None)
        


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)

File: cpp_solutions/706. Design HashMap.cc
class MyHashMap {
public:
    map<int, int> maps;
    MyHashMap() {
        maps.clear();
    }
    
    void put(int key, int value) {
        if (maps.find(key) == maps.end()) {
            maps.insert({key,value});
        } else {
            maps[key] = value;
        }
    }
    
    int get(int key) {
        if (maps.find(key)!=maps.end()) {
            return maps[key];
        } else {
            return -1;
        }
    }
    
    void remove(int key) {
        if (maps.find(key)!=maps.end()) {
            maps.erase(key);
        } 
    }
};

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap* obj = new MyHashMap();
 * obj->put(key,value);
 * int param_2 = obj->get(key);
 * obj->remove(key);
 */
